Feat/raw files in tasklist

underdarknl opened the pull request:
### Changes

Adds limited length in-line viewing of raw files on the task list. Still needs work to adhere to the designs: https://www.figma.com/file/QRL8085Z827bNdO0bIDdFL/KAT-%7C-Design-%7C-For-review?node-id=1%3A3&t=lZE8qsKdKNgKvk9Y-1, but @stephanie0x00 's security considerations have been applied.

### Issue link

Closes (https://github.com/minvws/nl-kat-coordination/issues/325

### Demo

![image](https://github.com/user-attachments/assets/1ca17ddc-37ab-4883-b4e9-2887c6245821)

### QA notes

Raw files should be visible when opening the expando-row on a task in the tasklist.
Boefjes wilt multiple raw files should show all.
Massive raw files should be limited to 1MB.
binary raw files will probably still break things.

### Code Checklist

<!--- Mandatory: --->

- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] I have checked the documentation and made changes where necessary.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

<!--- If applicable: --->

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

underdarknl commented:
Somewhere, we need to add some CSS that makes the pre inside the rawfiles list not be max-width: 100% but max-width: 100vw;

underdarknl commented:
![image](https://github.com/user-attachments/assets/664a3690-7df2-4d9e-b1cd-84c9f7a18b01)
Binary files do show up, however, they should be rendered as a hex display I'd say.

ammar92 commented:
I've fixed some errors, linting and language files. Moving it to QA

stephanie0x00 commented:
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [ ] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [ ] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
- Raw files are shown \o/
- Boefjes with multiple raw files show all raw files.


### What doesn't work:
- Tried to test a binary file upload, however none of the binary files tried would be parsed by OpenKAT, so still looking into this. 
- See 4 points below. 

### Bug or feature?:

1. Displaying of data could be more user friendly to enhance readability. 
   - There are two headers (yielded objects and yielded raw files). I believe Yielded objects only belongs on the Normalizers tab and this boefjes tasks page should only show the yielded raw files? 
   - The hashes of the files seem sort of out of nowhere and can throw off non-technical users. So maybe each new raw file should start with something like: "Raw file: X", to make more clear what is going on. 
   - The secure hash seems to be always signed by null. Is this a bug? 
![image](https://github.com/user-attachments/assets/26e8f15c-9d01-4c14-b29b-4ded9db67a86)



2. When expanding a row and then collapsing it again the interface seems to move a little.


3. The tasks list for boefjes shows often show a scroll bar instead of a code block. A code block makes it more user friendly. HTML does seem to get somewhat parsed (not wrapped though), but JSON is just a one-liner without any formatting. DNS records data seems to be parsed as a code block as well (with the scroll bar). 

![image](https://github.com/user-attachments/assets/9625a2ef-7d65-4355-afc2-b1ab53469841)


4. The normalizer page seems to have some testing code 

![image](https://github.com/user-attachments/assets/17e4e712-80e1-48e9-a965-d8b0e4ec9758)

underdarknl commented:
> ### What doesn't work:
> 
>     * Tried to test a binary file upload, however none of the binary files tried would be parsed by OpenKAT, so still looking into this.
Any image url should return a binary file.
> 
>     * See 4 points below.
> 
> 
> ### Bug or feature?:
> 
>     1. Displaying of data could be more user friendly to enhance readability.
>        
>        * There are two headers (yielded objects and yielded raw files). I believe Yielded objects only belongs on the Normalizers tab and this boefjes tasks page should only show the yielded raw files?
Fixed, Now only shows relevant header

>        * The hashes of the files seem sort of out of nowhere and can throw off non-technical users. So maybe each new raw file should start with something like: "Raw file: X", to make more clear what is going on.
Fixed, Its a Code block now its also no londer above the content.his

>        * The secure hash seems to be always signed by null. Is this a bug?
No, it depends on the signing provider. Its no longer shown if its null, its now also a link to the signing provider if available.

>     2. When expanding a row and then collapsing it again the interface seems to move a little.
Not noticed this, but I did fix any eronious unneeded reloading if the expando row had already been filled.

>     3. The tasks list for boefjes shows often show a scroll bar instead of a code block. A code block makes it more user friendly. HTML does seem to get somewhat parsed (not wrapped though), but JSON is just a one-liner without any formatting. DNS records data seems to be parsed as a code block as well (with the scroll bar).
I've changed the width of the code block to be less wide, it was still wider than the expando row before that.

>     4. The normalizer page seems to have some testing code

Now fixed, it was a left over part of the html.

I'm in the process of adding a hex set of tabs that show the content as plaintext, prettyprinted json and binary/hex. The first two are now implemented. The latter requires a bit more JS. Currently all views are visible continously, those should however be hidden by the toggle/tab feature above it. This tab feature relies on the mentioned PR.

underdarknl commented:
Next to plain text view also with Json view, (if json parseable)
![image](https://github.com/user-attachments/assets/52be33f8-c04b-4496-b20c-d6b5d8fc8fd8)
and with HEX view if not json parseable.
![image](https://github.com/user-attachments/assets/07210b6f-269f-407a-a2e0-6da78fe66a9d)
